lib/matplotlib/cbook/__init__.py

```diff
@@ -1499,6 +1499,34 @@ def violin_stats(X, method, points=100, quantiles=None):
     return vpstats
 
 
+def pad_arrays(v, padval=np.nan):
+    """
+    Pad list of arrays of varying lengths to the same size with specified
+    value
+
+    Parameters
+    ----------
+    v : iterable
+        List of arrays to be padded to the largest len. All elements must
+        support iteration
+
+    padval : scalar, bool or NaN, defaul NaN
+        value to pad missing values with
+
+    Returns
+    -------
+    out : array
+        Array of input arrays padded to the same len by specified padval
+
+    Examples
+    --------
+    >>> a, b, c = pad_arrays([1,2,3,4], [1,2,3], [1])
+    """
+    if len(set([k.shape[0] for k in v])) == 1:
+        return v
+    return np.array(list(itertools.zip_longest(*v, fillvalue=padval))).T
```

Comment:
This is pretty obfuscated.  Is there a performance reason for this to be written like this?